### PR TITLE
ARROW-4754: [Java] Randomize port and retry binding server when bind fails

### PR DESCRIPTION
**Pull request by emkornfield**
Also, random cleanups to make intellij happy.

Should prevent reoccurences of  recent master build failure: https://travis-ci.org/apache/arrow/jobs/501310887

### CODE REVIEW

File: java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java [hunk 1]
```diff
@@ -94,22 +95,38 @@ public boolean isValid(byte[] token) {
...
+    // we see flakiness when we use the same port each time.
+    Location location = null;
+    IOException lastThrown = null;
+    Random random = new Random();
+    for (int x = 0; x < 3; x++) {
+      final int port = 49152 + random.nextInt(5000);
+      location = new Location("localhost", port);
+      lastThrown = null;
+      try {
+        server = new FlightServer(
```
Comment (praveenbingo) [inline java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java]: do we need to start the server for each test? can we do it once for the class and create new clients for each test?

**Review by praveenbingo — COMMENTED** [review]
@emkornfield thanks for fixing this :)

File: java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java (hunk 1 above)
Comment (emkornfield) [reply java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java]: In general I would prefer not to do that as long as server start is quick (which I think it should be).

**Comment by emkornfield** [comment]
I pushed the fix for JIRA-4755 to this PR as well (sorry if this is a randomization).

File: java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java (hunk 1 above)
Comment (praveenbingo) [reply java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java]: hmm ok..would that not be the actual real world scenario though? (should we have another test for that else?)

**Comment by wesm** [comment]
@praveenbingo any more comments on this?

File: java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java (hunk 1 above)
Comment (emkornfield) [reply java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java]: Yes, I think another test would be better for that.  I haven't been involved in flight development but I'll leave up to the community members that have to open a JIRA if one already isn't open for more integration testing like scenarios.

Comment (praveenbingo) [reply java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java]: Ok i am not sure i agree :) but either ways the test is in much better shape than it was..we can refactor more later. 

Thanks again !

**Comment by praveenbingo** [comment]
@wesm No, please help in merging this.

**Review by praveenbingo — APPROVED** [review]
+1 LGTM.

**Comment by praveenbingo** [comment]
@pravindra - please help merge..

**Pull request closed without merging**